spec/services/billing_entities/update_invoice_grace_period_service_spec.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe BillingEntities::UpdateInvoiceGracePeriodService, type: :service do
+  include ActiveJob::TestHelper
+  subject(:update_service) { described_class.new(billing_entity:, grace_period:) }
+
+  let(:billing_entity) { create(:billing_entity) }
+  let(:organization) { billing_entity.organization }
+  let(:customer) { create(:customer, organization:) }
+  let(:grace_period) { 2 }
+
+  describe "#call" do
+    # let(:invoice_to_be_finalized) do
+    #   create(:invoice, status: :draft, customer:, issuing_date: DateTime.parse("19 Jun 2022").to_date, organization:)
+    # end
+    #
+    # let(:invoice_to_not_be_finalized) do
+    #   create(:invoice, status: :draft, customer:, issuing_date: DateTime.parse("21 Jun 2022").to_date, organization:)
+    # end
```

Comment:
I know this is commented code for later, however, I would expect the code to be correct... should the created invoices be linked to the billing entity here?

Reply:
it will be connected, because when we create invoice, passing organization, we assign the invoice to the default billing_entity for this organization. Though, you're right, it's better to do it explicitly